docs/boogie-code.md

```diff
@@ -38,6 +38,42 @@ void assume(bool v) {
 }
 ````
 SMACK interprets the `@` symbol by inserting the value of the C variable `v`.
+Since the arguments of variadic functions may be promoted, SMACK allows users
+to append a format character to the `@` symbol as an indicator that the argument,
+rather than its promoted form, should be used in the format string. The format
+characters are given in the following table along with their corresponding C
+types and their effective types within SMACK:
+
+| Format | Original C Type | SMACK Type |
+|--------|-----------------|------------|
+| c      | char            | i8         |
```

Comment:
I can also see this being written like:
- `'c'/'b'/'B'` correspond to the char/signed char/unsigned char types respectively and correspond to the i8 type.
- ...

I don't have a real preference between this and the table.

Reply:
I don't think we should be exposing SMACK type here. For example, if you enable `--bit-precise`, it would be bv8 instead of i8. But we should have information regarding which character matches which type. I think though we should remove the table and have that information be in a sentence, such as: `We allow the following format characters for the respective C types: `c` (char), `b` (signed char)...`

Reply:
Btw, do these characters match the typical C printf format characters? It would be good if they would. So then you should also say "We allow the following format characters (inspired by C) for the respective C types:..."

Reply:
I think Python based these characters on the `printf` format characters. Direct `printf` versions would look like: `@c` -> `char`, `@h(d?)` -> `short`, `@d` ->`int` (`int` doesn't have a length specifier apparently) and `@f` -> `float` (technically `double` according to `printf`). The differences in our format are using `@i` instead of `@d` for `int` and `short` being `@h` instead of `@hd`. I think "inspired by C" would be fine for this. 

I can change the doc to a sentence.

Reply:
Well, then maybe say "inspired by Python" :)

Reply:
Or even better "inspired by C and Python"